ansible/nats/install.yml

```diff
@@ -0,0 +1,283 @@
+            issuerRef:
+              name: ca-issuer
+              kind: Issuer
+            commonName: nats.{{ ns }}.svc.cluster.local
+            dnsNames:
+              - localhost
+              - "{{ ui_wildcard_fqdn }}"
+              - "{{ vice_wildcard_fqdn }}"
+              - "{{ vice_base_domain }}"
+              - "*.nats.{{ ns }}.svc.cluster.local"
+              - "*.nats.{{ ns }}.svc"
+              - "*.nats.{{ ns }}"
+              - "*.nats"
+              - "*.nats-headless.{{ ns }}.svc.cluster.local"
+              - "*.nats-headless.{{ ns }}.svc"
+              - "*.nats-headless.{{ ns }}"
+              - "*.nats-headless"
+              - nats.{{ ns }}.svc.cluster.local
+              - nats.{{ ns }}.svc
+              - nats.{{ ns }}
+              - nats
+
+    - name: create the nats values file from inline content
+      ansible.builtin.copy:
+        content: |
+          tlsCA:
+            enabled: true
+            secretName: selfsigned-ca
+            dir: /etc/nats/tls/ca/
+            ca: ca.crt
+
+          service:
+            enabled: true
+            ports:
+              nats:
+                enabled: true
+              leafnodes:
+                enabled: true
+              websocket:
+                enabled: true
+              cluster:
+                enabled: true
+
+          config:
+            cluster:
+              enabled: true
+              noAdvertise: true
+              port: 6222
+              replicas: 4
+              routeURLs:
+                useFQDN: false
+                k8sClusterDomain: cluster.local
+              tls:
+                enabled: true
+                secretName: nats-server-tls
+                dir: /etc/nats/tls/cluster/
+                cert: tls.crt
+                key: tls.key
+            nats:
+              port: 4222
+              tls:
+                enabled: true
+                secretName: nats-client-tls
+                dir: /etc/nats/tls/nats/
+                cert: tls.crt
+                key: tls.key
+            leafnodes:
+              enabled: true
+              noAdvertise: true
+              port: 7422
+              tls:
+                enabled: true
+                secretName: nats-client-tls
+                dir: /etc/nats/tls/leafnodes/
+                cert: tls.crt
+                key: tls.key
+            websocket:
+              enabled: true
+              port: 8080
+              tls:
+                enabled: true
+                secretName: nats-server-tls
+                dir: /etc/nats/tls/websocket/
+                cert: tls.crt
+                key: tls.key
+        dest: "./values.yaml"
+
+    - name: Install nats helm chart
+      environment:
+        KUBECONFIG: "{{ lookup('env', 'KUBECONFIG') }}"
+      kubernetes.core.helm:
+        name: nats
+        chart_ref: nats/nats
+        release_namespace: "{{ ns }}"
+        wait: true
+        values_files:
+          - "./values.yaml"
```

Comment:
It looks like the `values` parameter takes complex content if you wanted to install this without generating a local file as well.